packages/rum/src/domain/record/record.ts

```diff
@@ -53,34 +67,33 @@ export function record(options: RecordOptions): RecordAPI {
 
   function flushMutations() {
     shadowRootsController.flush()
-    flushMutationsFromObservers()
+    mutationObserver.flush()
   }
 
-  const { stop: stopObservers, flush: flushMutationsFromObservers } = initObservers(configuration, {
-    lifeCycle: options.lifeCycle,
-    elementsScrollPositions,
-    inputCb: emitAndComputeStats,
-    mediaInteractionCb: emitAndComputeStats,
-    mouseInteractionCb: emitAndComputeStats,
-    mousemoveCb: emitAndComputeStats,
-    mutationCb: emitAndComputeStats,
-    scrollCb: emitAndComputeStats,
-    styleSheetCb: emitAndComputeStats,
-    viewportResizeCb: emitAndComputeStats,
-    frustrationCb: emitAndComputeStats,
-    focusCb: emitAndComputeStats,
-    visualViewportResizeCb: emitAndComputeStats,
-    viewEndCb: (viewEndRecord) => {
+  const recordIds = initRecordIds()
+  const mutationObserver = initMutationObserver(emitAndComputeStats, configuration, shadowRootsController, document)
+  const observers = [
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏ what about being explicit with the types here. They are inferred quite well, but I think it could help reading and understanding the code.

```suggestion
  const observers: Observer[] = [
```

with `Observer` something like `type Observer = { stop: () => void }`.

Then you could also be explicit in the return type of all the `init*Observer` methods (Seems like some of them wouldl return `FlushableObserver`, or something similar?)

wdyt?

Reply:
I wouldn't mind naming the type explicitly, although `Observer` might not be great? It comes from RRWeb, but we don't *need* to stick with it.

* We often call those kind of function `trackSomething` (ex: `trackViews`, `trackActions`...), and sometimes `somethingObservable` (ex: `xhrObservable`...)
* We have a larger concept of things that can be "stopped"
* I often confuse `Observer` with `Observable`

Maybe we could ditch the "Observer" naming, and use "track"? (Or "observable", but it we don't use `Observable` objects there yet).

Maybe we could have a `type Stoppable = { stop: () => void }` that we can reuse in other parts of the SDK?

Reply:
I agree it's confusing to have `Observer` and `Observable`, I like having consistent names "trackSomething()". However I'm not sure about `Stoppable`, what about something more related to the tracking concept like `type TrackingResult = { stop: () => void }`

Let see how the current `const observers: Observer[initObserver, ...]` would look like? 
- `const stoppables: Stoppable[] = [ trackSomething,...`,
- `const trackingResults: TrackingResult[] = [ trackSomething,...`
-  (Maybe) `const trackers: Tracker[] = [ trackSomething,...`

What do you think?
cc @bcaudan

Reply:
> However I'm not sure about Stoppable, what about something more related to the tracking

👍

> I like having consistent names `trackSomething()`

Looking at the code base, we are using different name patterns for similar usages `trackXXX`, `XXXCollection`, `XXXObserver`, `XXXObservable`, not sure there is clear distinction between all of them.

> - const trackingResults: TrackingResult[] = [ trackSomething,...
> - (Maybe) const trackers: Tracker[] = [ trackSomething,...

Given that the object we are retrieving is "actionable", I would lean toward `tracker` as the `result` suffix would more express to me a simple data structure (without business logic attached).

Reply:
> result suffix would more express to me a simple data structure (without business logic attached).

I agree! `Tracker` looks good to me, otherwise `Subscription` seems like a good fit too.

> Looking at the code base, we are using different name patterns for similar usages trackXXX, XXXCollection, XXXObserver, XXXObservable, not sure there is clear distinction between all of them.

Yeah, and that makes onboarding to the project difficult. It would be great to standardise these concepts. I tend to prefer well know names such as `Observer`, `Subscription`, `Subject` where possible but as long as it's consistent within the browser SDK it should be fine.